code/game/objects/items/blueprints.dm

```diff
@@ -216,6 +218,7 @@
 		A.setup(str)
 		A.contents += turfs
 		A.SetDynamicLighting()
+	A.has_gravity = old.has_gravity
```

Comment:
has_gravity(src) would return whether the area currently has gravity (including from a generator), not whether the area innately has it (such as lavaland or centcomm)